File: app/queue.py
from rq import Queue
from redis import Redis
from app.config import settings

redis_conn = Redis.from_url(settings.REDIS_URL)
queue = Queue(settings.RQ_QUEUE_NAME, connection=redis_conn)


def enqueue_submission(submission_id: int) -> None:
	queue.enqueue("worker.worker.judge_submission", submission_id)

File: server/src/queue.ts
import { Queue } from 'bullmq'
import { Redis } from 'ioredis'

const connection = new Redis(process.env.REDIS_URL || 'redis://redis:6379/0', {
	maxRetriesPerRequest: null,
	enableReadyCheck: false
})
export const judgeQueue = new Queue('judging', { connection })

export async function enqueueSubmission(submissionId: string) {
	await judgeQueue.add('judge', { submissionId })
}



